src/webgpu/shader/execution/builtin/inversesqrt.spec.ts

```diff
@@ -0,0 +1,111 @@
+export const description = `
+Execution Tests for the 'inverseSqrt' builtin function
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { assert } from '../../../../common/util/util.js';
+import { GPUTest } from '../../../gpu_test.js';
+import { f32, f32Bits, Scalar, TypeF32, u32 } from '../../../util/conversion.js';
+
+import { Case, Config, kBit, run, ulpThreshold } from './builtin.js';
+
+export const g = makeTestGroup(GPUTest);
+
+// Calculates the linear interpolation between two values of a given ratio
+function lerp(min: number, max: number, t: number) {
+  if (!Number.isFinite(min) || !Number.isFinite(max) || !Number.isFinite(t)) {
+    return Number.NaN;
+  }
+  if (min === max) {
+    return min;
+  }
+
+  if (min > max) {
+    const temp = min;
+    min = max;
+    max = temp;
+  }
+
+  if (t > 1) {
+    return max;
+  }
+
+  if (t < 0) {
+    return min;
+  }
+
+  return min + t * (max - min);
```

Comment:
I wonder if this is good numerically.  For example, if max and min are both large magnitude and close to each other, then you get catastrophic cancellation.
(Subtracting to float numbers that agree to k bits means you lose k bits of precision, generally).

My next thought was to suggest:


       (min * (1-t)  + max * t)

Then I googled "numerically stable linear interpolation" and found someone who has thought about this way more deeply. There is a relatively simple solution that has nice properties.

http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p0811r2.html 

Please use the suggested algorithm.